### PR TITLE
implement read/unread in service and client

### PR DESCRIPTION
**Pull request by songgao**
Not exactly what's in the new design doc but I'll update this to new design in a separate PR.

r? @mmaxim @patrickxb @maxtaco Thanks!

### CODE REVIEW

File: protocol/avdl/chat_local.avdl [hunk 1]
```diff
@@ -56,17 +57,28 @@ protocol chatLocal {
...
 
   chat1.InboxView getInboxLocal(union { null, chat1.Pagination } pagination);
-  ThreadView getThreadLocal(chat1.ConversationID conversationID, union { null, chat1.Pagination } pagination);
+  ThreadView getThreadLocal(chat1.ConversationID conversationID, boolean markAsRead, union { null, chat1.Pagination } pagination);
   void postLocal(chat1.ConversationID conversationID, MessagePlaintext messagePlaintext);
   chat1.ConversationID newConversationLocal(chat1.ConversationIDTriple conversationTriple);
 
   chat1.ConversationID getOrCreateTextConversationLocal(string tlfName, string topicName, chat1.TopicType topicType);
-  array<Message> getMessagesLocal(MessageSelector selector);
+  array<ConversationMessagesLocal> getMessagesLocal(MessageSelector selector);
```
Comment (songgao) [inline protocol/avdl/chat_local.avdl]: I've been experimenting myself with different versions of the interface between CLI and the service. I think the current `getMessagesLocal` should cover most message fetching logics from CLI. Let me know if this looks reasonable!

File: go/client/chat_utils.go [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+	}
+
+	m[uid] = ret.Username
+	return ret.Username, err
+}
+
+// parseDurationExtended is like time.ParseDuration, but adds "d" unit. "1d" is
+// one day, defined as 24*time.Hour. Only whole days are supported for "d"
+// unit, but it can be followed by smaller units, e.g., "1d1h".
+func parseDurationExtended(s string) (d time.Duration, err error) {
```
Comment (maxtaco) [inline go/client/chat_utils.go]: it seems like with @malgorithms's new proposal of how the chat API will work, we'll want all of this stuff in the service. We should discuss this today if possible in person.

File: protocol/avdl/chat_local.avdl (hunk 1 above)
Comment (maxtaco) [inline protocol/avdl/chat_local.avdl]: 👍

File: go/client/chat_utils.go (hunk 2 above)
Comment (songgao) [inline go/client/chat_utils.go]: hmm.. this is CLI specific rendering. Do we want to move all CLI rendering related stuff to service?

Comment (maxtaco) [inline go/client/chat_utils.go]: Maybe, still wondering what to do based on how to set up the code.

Comment (songgao) [inline go/client/chat_utils.go]: We could do some in-RPC framing like what's used in gregor OOBM notification? So we'd only have a ChatAPI RPC, which accepts a JSON and returns a JSON. We can use the same API from the CLI, but at the same time other applications can use it through `kbc --json`

Comment (maxtaco) [inline go/client/chat_utils.go]: Right, was thinking along those lines, but there are lots of subtle usability problems that come up with that, talking them through with @malgorithms. For instance: enums are they ints or strings?  Or key names, are they snake-cased or command-line style or camelcase?

**Comment by maxtaco** [comment]
👍 after CI green (Circle is currently broken).

**Pull request merged**